DPIA knowledge model

vildead opened the pull request:
work in progress on knowledge model and template section

Commit:
starting on dpia knowlegde model

Commit:
KM report first draft

Commit:
fix typos

Commit:
motivations with DSW

Commit:
Slight rewrite of Abstract

Commit:
Slight rewrite of Introduction, part 1.

Commit:
Slight rewrite of Introduction, part 2.

Commit:
Slight rewrite of DPIA in DSW

Commit:
Slight rewrite of DPIA KM, part 1.

Commit:
Slight rewrite of DPIA KM, part 2.

Commit:
Slight rewrite of DPIA KM, part 2; more...

Commit:
Slight rewrite of output template

Commit:
Slight rewrite of DPIA projects

vildead committed:
extend of 'DPIA in DSW' section

vildead committed:
add link to CNIL PIA tool

vildead committed:
update of section 'Creating a DPIA Knowledge Model

- add comment on risk assessment
- minor adjustments

vildead commented on paper.md:
```diff
@@ -50,35 +50,61 @@ event: BH2021
 
+In order to have the possibility to export the DPIA Knowledge model in a  document format which follow the structure of the CNIL Template, we have set up the DPIA Document Template.  It allows the different responses of the DPIA knowledge model to be collected in one document that can be exported in differents formats, e.g., HTML, MS WORD Document, etc... We used the [Template Development Kit (TD ...
+
 ## DPIA Projects in DSW
```
@barrydjenaba could you please merge this section with previous one? I think this is not needed since the linking between KM and template can be described in previous section

vildead replied:
done

vildead commented on paper.md:
```diff
@@ -50,35 +50,61 @@ event: BH2021
 
 *TODO: Nene and Paulette - what was the process, how you worked, what is the result*
 
+In order to have the possibility to export the DPIA Knowledge model in a  document format which follow the structure of the CNIL Template, we have set up the DPIA Document Template.  It allows the different responses of the DPIA knowledge model to be collected in one document that can be exported in differents formats, e.g., HTML, MS WORD Document, etc... We used the [Template Development Kit (TD ...
```
@barrydjenaba could you please include also reason why we chose the CNIL template and if there are other templates which can be added? (which can be potentially populated using our KM)

vildead commented on paper.md:
```diff
@@ -50,35 +50,61 @@ event: BH2021
 
 ## DPIA Document Template
 
 *TODO: Nene and Paulette - what was the process, how you worked, what is the result*
```
@barrydjenaba you can delete this line ;)

vildead replied:
line deleted

vildead commented on paper.md:
```diff
@@ -50,35 +50,61 @@ event: BH2021
 
 *TODO: Nene and Paulette - what was the process, how you worked, what is the result*
 
+In order to have the possibility to export the DPIA Knowledge model in a  document format which follow the structure of the CNIL Template, we have set up the DPIA Document Template.  It allows the different responses of the DPIA knowledge model to be collected in one document that can be exported in differents formats, e.g., HTML, MS WORD Document, etc... We used the [Template Development Kit (TD ...
```
the second sentence "It allows the different responses... " can be imho deleted. It is describing purely what a DSW can do. I suggest to include more template specific information - does our KM contain all the information required by CNIL? Are there any answers not used by the template? Are there any answers which are aggregated? How is the administrative infromation mapped to the template.... etc..something on this topic - doenst have to answers all :)

vildead replied:
I have added small information on the template properties. 
@froggypaule Could you please take a quick look at this section and my proposed content above - if there is anything coming in your mind it would be great. I was not involved in the template implementation so my knowledge here is very limited. Also, if you dont have anything to add, its totally fine. I would say we can give it a go :)

froggypaule replied:
Hello @vildead here we go. I did a read through. (I did a review process, forgot how to properly review git commits!!)
Thanks for the opportunity, cheers!

vildead reviewed (commented):
Almost there. Still some minor changes needed imho

Commit:
dpia document template

vildead committed:
minor updates Creating a DPIA Knowledge Model section

vildead committed:
major update of DPIA Document Template

- sentences describing general DSW template description are removed
- reason for selection of CNIL templated is explained

vildead committed:
remove section DPIA Projects in DSW

- general description of how KM-template works in DSW is imho not needed
- if there is anything specific for our KM-template interaction, it can
be included in previous section

froggypaule commented on paper.md:
```diff
@@ -76,19 +76,21 @@ Finally, as both data management plan and data protection impact assessment are
 ## Creating a DPIA Knowledge Model
 
-*TODO: (Pinar), Vilém, Nene, and Paulette - what was the process, how you worked, what is the result*
+The DPIA knowledge model is mainly inspired by the several existing documents, namely the [CNIL DPIA document template](https://www.cnil.fr/sites/default/files/atoms/files/cnil-pia-2-en-templates.pdf), the DPIA document template used by the Luxembourg Centre for Systems Biomedicine and [ICO guidance](https://ico.org.uk/for-organisations/guide-to-data-protection/guide-to-the-general-data-protectio ...
```
- ...'is mainly inspired by several...': remove the before several
- and the ICO...: add the before ICO

froggypaule commented on paper.md:
```diff
@@ -76,19 +76,21 @@ Finally, as both data management plan and data protection impact assessment are
+The DPIA knowledge model is mainly inspired by the several existing documents, namely the [CNIL DPIA document template](https://www.cnil.fr/sites/default/files/atoms/files/cnil-pia-2-en-templates.pdf), the DPIA document template used by the Luxembourg Centre for Systems Biomedicine and [ICO guidance](https://ico.org.uk/for-organisations/guide-to-data-protection/guide-to-the-general-data-protectio ...
 
-The DPIA knowledge model aims to help in managing the data protection risks of a research project by identifying and analyzing those risks and the solutions implemented to mitigate them. This knowledge model is mainly inspired by the [CNIL Template](https://www.cnil.fr/sites/default/files/atoms/files/cnil-pia-2-en-templates.pdf), the DPIA template from the Luxembourg Centre for Systems Biomedicin ...
+The first chapter collects administrative information and description of method followed for during project's DPIA. Following chapters contain questions related to:
```
The first chapter collects administrative information and description of the method(s) followed to establish the project's DPIA. Following chapters contain questions related to:

froggypaule commented on paper.md:
```diff
@@ -76,19 +76,21 @@ Finally, as both data management plan and data protection impact assessment are
-* The proportionality and necessity of the processing;
-* The measures taken to protect the rights of the subjects;
-* The identification, analyzis, and evaluation of the data risks factors and the measures and controls to mitigate those risks.
+Although a risk assessment is very important part of a DPIA, our knowledge model is addressing it very briefly only by questions on general controls which are or will be in place. As risks generally depend on the used infrastructure and involved institutional units, they are globally applicable to all the processing under same institute. We propose process, where the risk assessment matrix can be ...
```
Although a risk assessment is **a** very important part of **the** DPIA

froggypaule replied:
our knowledge model is addressing it very briefly by questions on general controls

froggypaule replied:
As risks generally depend on the infrastructure used and the  institutional units involved, they would apply globally to all the processing under their authority.

froggypaule replied:
We propose a process where the risk assessment matrix can be generated by in-house or external professionals using dedicated tools (e.g. [Monarc](https://www.monarc.lu/)) and reused for each DPIA as a resource.

froggypaule replied:
For this purpose, our knowledge model includes **a** question to provide link to such a document.

froggypaule reviewed (commented):
my usual being pedantic with the writing ;)

froggypaule commented on paper.md:
```diff
@@ -68,7 +68,7 @@ To support and disseminate the use of our integration work, we also planned to c
 Writing a data protection impact assessment for a scientific project is a challenging task. Input from multiple key groups is required to obtain complete and precise information. In addition to data stewards, which are commonly the ones in charge of the information collection since their knowledge overlaps all other involved domains, the process involves researchers themselves bringing knowledge  ...
 
-DPIA is commonly written using spreadsheet or text based templates with questions and predefined answers. This approach suffers from the absence of branching logic, overwhelming and poorly structured content and lack of support for collaborative editing. Furthermore, subsequent updates of the template make the whole process laborious and unnecessarily complex. Dedicated tools (e.g. PIA [citation  ...
+DPIA is commonly written using spreadsheet or text based templates with questions and predefined answers. This approach suffers from the absence of branching logic, overwhelming and poorly structured content and lack of support for collaborative editing. Furthermore, subsequent updates of the template make the whole process laborious and unnecessarily complex. Dedicated tools (e.g. [Processing Im ...
```
This approach suffers from the absence of **a** branching logic, overwhelming and poorly structured content, and **a** lack of support for collaborative editing.

froggypaule replied:
[Processing Impact Assessment](https://www.cnil.fr/en/open-source-pia-software-helps-carry-out-data-protection-impact-assesment) from **the** Commission Nationale de l'Informatique et des Libertés (CNIL)) are also available and they do bring substantial help but  in general they lack the ability of a full customization of the template.

froggypaule replied:
**A** DPIA is commonly written using spreadsheet

froggypaule reviewed (commented):
idem

froggypaule commented on paper.md:
```diff
@@ -66,14 +66,13 @@ To support and disseminate the use of our integration work, we also planned to c
 # DPIA in DSW
 
-*TODO: (Pinar) and Vilém - motivation for having DPIA KM+template in DSW, core idea and advantages*
+Writing a data protection impact assessment for a scientific project is a challenging task. Input from multiple key groups is required to obtain complete and precise information. In addition to data stewards, which are commonly the ones in charge of the information collection since their knowledge overlaps all other involved domains, the process involves researchers themselves bringing knowledge  ...
```
In addition to data stewards, which are commonly in charge of the information collection since their expertise overlaps all other concerned domains, the process  involves researchers themselves bringing their own knowledge about the data involved and the related scientific processes. Data managers and IT staff who are knowledgeable about the local IT infrastructure, and external collaborators may also report on the shared data processing tasks.

froggypaule replied:
Finally, legal representatives (e.g. data protection officer**s**) validate the resulting document using existing contracts and institutional policies.

froggypaule commented on paper.md:
```diff
@@ -66,14 +66,13 @@ To support and disseminate the use of our integration work, we also planned to c
+Writing a data protection impact assessment for a scientific project is a challenging task. Input from multiple key groups is required to obtain complete and precise information. In addition to data stewards, which are commonly the ones in charge of the information collection since their knowledge overlaps all other involved domains, the process involves researchers themselves bringing knowledge  ...
 
-DAISY aims at recording all DPIA relevant information on the processing of sensitive data. DPIA is usually set up using a template that researchers customise to suit their research project. The idea is therefore to use the flexibility of the DSW tool to set up in DSW a knowledge model which will allow users to create a DPIA project by adding directly the relevant information contained in DAISY (G ...
+DPIA is commonly written using spreadsheet or text based templates with questions and predefined answers. This approach suffers from the absence of branching logic, overwhelming and poorly structured content and lack of support for collaborative editing. Furthermore, subsequent updates of the template make the whole process laborious and unnecessarily complex. Dedicated tools (e.g. PIA [citation  ...
```
did comment in next commit

froggypaule commented on paper.md:
```diff
@@ -66,14 +66,13 @@ To support and disseminate the use of our integration work, we also planned to c
-Advantages:
-* Writing a DPIA in DSW offers the possibility for the user to be guided through the process with recommandations and questions following a well-defined hierarchical structure.
-* Also, it allows exporting the DPIA in a format and structure of choice (e.g. the CNIL template): that process is both machine- and human-readable.
+DSW offers variety of needed functionalities for DPIA document generation. It provides user access and project visibility control, which are necessary features since the DPIA content is often confidential. DSW supports creation of versioned structured questionnaire with a complex branching logic. Questions can be interlinked, accompanied by extensive explanation, hints, predefined answers and gui ...
```
DSW offers **a** variety of functionalities needed for DPIA document generation.

froggypaule replied:
It controls user access and project visibility control, which are necessary features since the DPIA content is often confidential

froggypaule replied:
DSW supports **the** creation of **a** versioned structured questionnaire with a complex branching logic.

froggypaule replied:
Questions can be interlinked, accompanied by extensive explanation**s**, hints, predefined answers and guidance; it may be enriched by links to external sources (e.g. specific GDPR content) or integrations to local or external vocabularies. The smooth collaboration is possible thanks to **the** ability to add comments on each question.

vildead committed:
addressing review comments from Paulette

MarekSuchanek commented:
- We usually use "Knowledge Model" capitalized, but it can also be in lower case of course... just we should make it consistent across the report.
- Could you also add link to the repo for the DPIA template where suitable (no need to change the text, just where "DPIA template" is mentioned - make it a link? https://github.com/ds-wizard/dpia-template

It looks very good, thanks - can be the then merged?

vildead committed:
capitalizing 'Knowledge Model'

vildead committed:
adding links to DPIA template

vildead commented:
> * We usually use "Knowledge Model" capitalized, but it can also be in lower case of course... just we should make it consistent across the report.
> * Could you also add link to the repo for the DPIA template where suitable (no need to change the text, just where "DPIA template" is mentioned - make it a link? https://github.com/ds-wizard/dpia-template
> 
> It looks very good, thanks - can be the then merged?

@MarekSuchanek thanks for the review! I have addressed your comments in the two new commits. 
I believe you can merge.
